lib/checks/canMultiplyLikeTermConstantNodes.js

```diff
@@ -0,0 +1,63 @@
+const Node = require('../node');
+const NodeType = require('../node/Type');
+
+// Returns true if node is a multiplication of constant power nodes
+// where you can combine their exponents, e.g. 10^2 * 10^4 * 10 can become 10^7
+
+function canMultiplyLikeTermConstantNodes(node) {
+  if (!Node.Type.isOperator(node) || node.op !== '*') {
+    return false;
+  }
+  const args = node.args;
+  if (!args.every(n => isPowerTerm(n))) {
+    return false;
+  }
+  const constantTermList = node.args.map(n => createPowerNode(n));
+  const firstTerm = constantTermList[0];
+  const restTerms = constantTermList.slice(1);
+  // they're considered like terms if they have the same base value
+  return restTerms.every(term => getBaseValue(firstTerm) === getBaseValue(term));
+}
+function createPowerNode(node) {
+  if (NodeType.isOperator(node)) {
+    if (node.op === '^') {
+      const constantNode = node.args[0];
+      if (!NodeType.isConstant(constantNode)) {
+        throw Error('Expected constant term, got ' + constantNode);
+      }
+      node.base = constantNode;
```

Comment:
I don't think it's a great idea to add extra fields to the mathjs node like that

How about this: make two functions that are given a node which is a constant or a constant with an exponent

one returns the base (or the node itself) and the other returns the exponent (or the constant 1 if there's no exponent)